Improve the key change menu and set a few keybindings to be debug only.

red-001 opened the pull request:
See #3846. This also partially fixes #3847. 

I have added more keys to the key change menu, fixed a bug that stop the  keybinding for `forwards` from being changed if the warning about a key already being in use was visible and added a warning if the user tried to assign the `escape` to a keybinding. I set some keybindings to only have a key set to them by default in debug builds.
![menu](https://cloud.githubusercontent.com/assets/12450071/13950943/d86e5714-f025-11e5-80c9-3aa7d49523d3.png)

kilbith commented on src/guiKeyChangeMenu.cpp:
```diff
@@ -396,23 +405,30 @@ void GUIKeyChangeMenu::add_key(int id, const wchar_t *button_name, const std::st
+	this->add_key(GUI_ID_KEY_SCREENSHOT_BUTTON, wgettext("Screenshot"),       "keymap_screenshot");
+	this->add_key(GUI_ID_KEY_RANGE_BUTTON,      wgettext("Range select"),     "keymap_rangeselect");
+	this->add_key(GUI_ID_KEY_INCREASE_RANGE, wgettext("Increase veiw range"), "keymap_increase_viewing_range_min");
+	this->add_key(GUI_ID_KEY_DECREASE_RANGE, wgettext("Decrease veiw range"), "keymap_decrease_viewing_range_min");
```
Typos (`veiw`).

ObaniGemini commented:
`The Debug menu might also be nice, as well as camera toggle.`
(from @C1ffisme on https://github.com/minetest/minetest/issues/3846 )

est31 commented on src/defaultsettings.cpp:
```diff
@@ -76,11 +76,11 @@ void set_default_settings(Settings *settings)
-	settings->setDefault("keymap_quicktune_next", "KEY_END");
-	settings->setDefault("keymap_quicktune_dec", "KEY_NEXT");
-	settings->setDefault("keymap_quicktune_inc", "KEY_PRIOR");
+	settings->setDefault("keymap_print_debug_stacks", "");
```
this should depend on #if debug.

red-001 replied:
done.

red-001 committed:
Improve the key change menu and set a few keybindings to be debug only.

I have added more keys to the key change menu, fixed a bug that stop the  keybinding for `forwards` from being changed if the warning about a key already being in use was visible and added a warning if the user tried to assign the `escape` to a keybinding. I set some keybindings to only have a key set to them by default in debug builds.

red-001 committed:
Add a button that resets the all the keybindings to default.

red-001 committed:
Add grantme command

Wuzzy2 commented:
I just noticed I created a VERY similar PR:
https://github.com/minetest/minetest/issues/4537

I like the ordering of the used actions in my PR more. Maybe you can adapt this order and apply it to your PR? Then I would be willing to close my own PR.

Besides, is the strange “If this menu screws up (…)” message still relevant? It seems to be a relic of ancient times. ;-)

rubenwardy commented on src/defaultsettings.cpp:
```diff
@@ -76,11 +76,19 @@ void set_default_settings(Settings *settings)
 	settings->setDefault("tooltip_show_delay", "400");
 
 	// Some (temporary) keys for debugging
+	#ifdef NDEBUG
```
I'd prefer `DEBUG` here, and swapping the two blocks, just to keep from having an unnecessary negative. It's also more consistent with the above (`keymap_toggle_update_camera`)

rubenwardy replied:
also, what's the code style with indentation and preprocessor guards?

sfan5 replied:
`DEBUG` isn't defined anywhere, if you want to have block run in release use `#ifndef NDEBUG`.

sfan5 replied:
Also concerning code style, preprocessor statements are usually without indentation regardless of all other stuff.

rubenwardy replied:
Look at the block above here

rubenwardy replied:
Debug is used in the block above here, so needs to be changed if it never exists

rubenwardy reviewed (commented):
seems handy, will review more later

bigfoot547 commented:
> \<@Wuzzy2\>Besides, is the strange “If this menu screws up (…)” message still relevant? It seems to be a relic of ancient times. ;-) 

More like a nostalgic blast from the past. ;)

bigfoot547 commented on src/game.cpp:
```diff
@@ -192,6 +198,22 @@ struct LocalFormspecHandler : public TextDest {
+			}
+
+			if (fields.find("btn_exit") != fields.end()) {
+				errorstream << "server change regected" << std::endl;
```
Spelling

bigfoot547 reviewed (approved):
I like it buuuut....

Wuzzy2 commented:
Why did this PR have to die?
